From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: zador-blood-stained <zador-blood-stained@users.noreply.github.com>
Date: Tue, 24 Oct 2017 17:44:22 +0300
Subject: [ARCHEOLOGY] Enable USB keyboards in sunxi u-boot

> X-Git-Archeology: > recovered message: > Switch to AUTOBOOT_KEYED with <Space> and <Ctrl-C> as autoboot abort
> X-Git-Archeology: > recovered message: > methods
> X-Git-Archeology: > recovered message: > Increase sunxi bootdelay to 1
> X-Git-Archeology: > recovered message: > Closes #789
> X-Git-Archeology: - Revision dd48a784f5a202d84aa2dc8b5553a571c1866b11: https://github.com/armbian/build/commit/dd48a784f5a202d84aa2dc8b5553a571c1866b11
> X-Git-Archeology:   Date: Tue, 24 Oct 2017 17:44:22 +0300
> X-Git-Archeology:   From: zador-blood-stained <zador-blood-stained@users.noreply.github.com>
> X-Git-Archeology:   Subject: Enable USB keyboards in sunxi u-boot
> X-Git-Archeology:
> X-Git-Archeology: - Revision 5f4c4c77b28b4d15fd0165809ccfaed7af5c73e5: https://github.com/armbian/build/commit/5f4c4c77b28b4d15fd0165809ccfaed7af5c73e5
> X-Git-Archeology:   Date: Sat, 25 Nov 2017 15:11:13 +0300
> X-Git-Archeology:   From: zador-blood-stained <zador-blood-stained@users.noreply.github.com>
> X-Git-Archeology:   Subject: [WIP] Update sunxi u-boot to v2017.11
> X-Git-Archeology:
> X-Git-Archeology: - Revision 9363e3ec4d3a3234c5bdab6b7a1e3fd606b68102: https://github.com/armbian/build/commit/9363e3ec4d3a3234c5bdab6b7a1e3fd606b68102
> X-Git-Archeology:   Date: Mon, 14 Jan 2019 16:52:22 -0500
> X-Git-Archeology:   From: Martin Ayotte <martinayotte@yahoo.ca>
> X-Git-Archeology:   Subject: u-boot v2018.11 migration + tons of patches touchups
> X-Git-Archeology:
> X-Git-Archeology: - Revision c474e35fe1fc795154baf88dec36cc01c6a5b567: https://github.com/armbian/build/commit/c474e35fe1fc795154baf88dec36cc01c6a5b567
> X-Git-Archeology:   Date: Fri, 12 Apr 2019 20:29:43 +0200
> X-Git-Archeology:   From: Igor Pecovnik <igor.pecovnik@gmail.com>
> X-Git-Archeology:   Subject: [ sunxi64 ] Moved A64 boards to mainline u-boot 2019.04 Tested basic functions with eMMC install on A64 Notebooks and Olinuxino
> X-Git-Archeology:
> X-Git-Archeology: - Revision 5e8cbd8e6aaa87ee03d08c761d5e765e81c4c686: https://github.com/armbian/build/commit/5e8cbd8e6aaa87ee03d08c761d5e765e81c4c686
> X-Git-Archeology:   Date: Fri, 17 Apr 2020 23:44:35 +0200
> X-Git-Archeology:   From: Igor Pecovnik <igorpecovnik@users.noreply.github.com>
> X-Git-Archeology:   Subject: Switch sunxi and sunxi64 to u-boot v2020.04 (#1894)
> X-Git-Archeology:
> X-Git-Archeology: - Revision 8277822addfb690716f431c2f29a59effe7dd16f: https://github.com/armbian/build/commit/8277822addfb690716f431c2f29a59effe7dd16f
> X-Git-Archeology:   Date: Tue, 06 Apr 2021 11:42:56 +0200
> X-Git-Archeology:   From: Igor Pecovnik <igorpecovnik@users.noreply.github.com>
> X-Git-Archeology:   Subject: Bump Allwinner u-boot to 2021.04 (#2745)
> X-Git-Archeology:
> X-Git-Archeology: - Revision 3cb7175c667da480e59ef5b9bb23e238b4955cdf: https://github.com/armbian/build/commit/3cb7175c667da480e59ef5b9bb23e238b4955cdf
> X-Git-Archeology:   Date: Thu, 15 Sep 2022 11:08:20 +0200
> X-Git-Archeology:   From: Igor Pecovnik <igorpecovnik@users.noreply.github.com>
> X-Git-Archeology:   Subject: Upgrade Allwinner boot loader to 2022.08 (#4168)
> X-Git-Archeology:
> X-Git-Archeology: - Revision 38db7daaf700a985036bb3d00be21c50f71e731c: https://github.com/armbian/build/commit/38db7daaf700a985036bb3d00be21c50f71e731c
> X-Git-Archeology:   Date: Thu, 10 Nov 2022 13:24:06 +0100
> X-Git-Archeology:   From: Markus Hoffrogge <mhoffrogge@gmail.com>
> X-Git-Archeology:   Subject: Fix U-Boot SUNXI enable-autoboot-keyed.patch for U-Boot v2022.07 [AR-1322] (#4403)
> X-Git-Archeology:
> X-Git-Archeology: - Revision f6a09def35e647a5442e0f92f399485be29f19a9: https://github.com/armbian/build/commit/f6a09def35e647a5442e0f92f399485be29f19a9
> X-Git-Archeology:   Date: Thu, 10 Nov 2022 21:49:36 +0100
> X-Git-Archeology:   From: Igor Pecovnik <igorpecovnik@users.noreply.github.com>
> X-Git-Archeology:   Subject: Moving patches to per board and removing obsolete (#4409)
> X-Git-Archeology:
---
 arch/arm/Kconfig | 2 ++
 boot/Kconfig     | 3 ++-
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/arch/arm/Kconfig b/arch/arm/Kconfig
index 8a1e223422..1a41215ebf 100644
--- a/arch/arm/Kconfig
+++ b/arch/arm/Kconfig
@@ -1123,6 +1123,8 @@ config ARCH_SUNXI
 	select USB_KEYBOARD if DISTRO_DEFAULTS && USB_HOST
 	select USB_STORAGE if DISTRO_DEFAULTS && USB_HOST
 	select SPL_USE_TINY_PRINTF
+	imply AUTOBOOT_KEYED
+	imply AUTOBOOT_KEYED_CTRLC 
 	select USE_PREBOOT
 	select SYS_RELOC_GD_ENV_ADDR
 	imply BOARD_LATE_INIT
diff --git a/boot/Kconfig b/boot/Kconfig
index ad035695a4..5ce54eefcf 100644
--- a/boot/Kconfig
+++ b/boot/Kconfig
@@ -1254,7 +1254,7 @@ config AUTOBOOT_FLUSH_STDIN
 config AUTOBOOT_PROMPT
 	string "Autoboot stop prompt"
 	depends on AUTOBOOT_KEYED
-	default "Autoboot in %d seconds\\n"
+	default "Autoboot in %d seconds, press <Space> to stop\\n"
 	help
 	  This string is displayed before the boot delay selected by
 	  CONFIG_BOOTDELAY starts. If it is not defined	there is no
@@ -1309,6 +1309,7 @@ config AUTOBOOT_DELAY_STR
 config AUTOBOOT_STOP_STR
 	string "Stop autobooting via specific input key / string"
 	depends on AUTOBOOT_KEYED && !AUTOBOOT_ENCRYPTION
+	default " "
 	help
 	  This option enables stopping (aborting) of the automatic
 	  boot feature only by issuing a specific input key or
-- 
Armbian

